nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-value-providers/src/main/java/org/apache/nifi/stateless/parameter/aws/SecretsManagerParameterValueProvider.java

```diff
@@ -0,0 +1,208 @@
+
+    public static final PropertyDescriptor AWS_CONFIG_FILE = new PropertyDescriptor.Builder()
+            .displayName("AWS Configuration File")
+            .name("aws-configuration-file")
+            .required(false)
+            .defaultValue("./conf/bootstrap-aws.conf")
+            .description("Location of the bootstrap-aws.conf file that configures the AWS credentials.  If not provided, the default AWS credentials will be used.")
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    private final ObjectReader objectReader = new ObjectMapper().reader();
+
+    private final Set<String> supportedParameterNames = new HashSet<>();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private AWSSecretsManager secretsManager;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    protected void init(final ParameterValueProviderInitializationContext context) {
+        super.init(context);
+
+        this.descriptors = Collections.unmodifiableList(Arrays.asList(AWS_CONFIG_FILE));
+
+        final String awsBootstrapConfFilename = context.getProperty(AWS_CONFIG_FILE).getValue();
+        try {
+            this.secretsManager = this.configureClient(awsBootstrapConfFilename);
+        } catch (final IOException e) {
+            throw new IllegalStateException("Could not configure AWS Secrets Manager Client", e);
+        }
+
+        cacheSupportedParameterNames();
+    }
+
+    private void cacheSupportedParameterNames() {
+        supportedParameterNames.clear();
+        final ListSecretsResult listSecretsResult = secretsManager.listSecrets(new ListSecretsRequest());
+        if (listSecretsResult != null) {
+            for (final SecretListEntry entry : listSecretsResult.getSecretList()) {
+                supportedParameterNames.add(entry.getName());
+            }
+        }
+    }
+
+    @Override
+    public boolean isParameterDefined(final String contextName, final String parameterName) {
+        return supportedParameterNames.contains(getSecretName(contextName, parameterName));
+    }
+
+    @Override
+    public String getParameterValue(final String contextName, final String parameterName) {
+        final String secretName = getSecretName(contextName, parameterName);
+        final GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest()
+                .withSecretId(secretName);
+        final GetSecretValueResult getSecretValueResult = secretsManager.getSecretValue(getSecretValueRequest);
+
+        if (getSecretValueResult == null) {
+            throw new IllegalArgumentException(String.format("Secret [%s] not found", secretName));
+        }
+        if (getSecretValueResult.getSecretString() != null) {
+            return getSecretValueResult.getSecretString();
+        } else {
+            throw new IllegalStateException("Binary secrets are not supported");
```

Comment:
Reading the documentation for [getSecretString() ](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/secretsmanager/model/GetSecretValueResult.html#getSecretString--), it sounds like the secret might have been stored in some other format. Recommend adjusting the error message to include the secret name:

```suggestion
            throw new IllegalStateException(String.format("Secret Name [%s] string value not found", secretName));
```